src/IO/H5/VolumeData.cpp

```diff
@@ -154,6 +162,308 @@ void append_element_extents_and_connectivity(
     }
   }
 }
+// Given a std::vector of grid_names, computes the number of blocks that exist
+// and also returns a std::vector of block numbers that is a one-to-one mapping
+// to each element in grid_names
+std::tuple<size_t, std::vector<size_t>, std::vector<std::vector<size_t>>>
+compute_and_organize_block_info(const std::vector<std::string>& grid_names) {
+  std::vector<size_t> block_number_for_each_element;
+  std::vector<std::vector<size_t>> sorted_element_indices;
+  block_number_for_each_element.reserve(grid_names.size());
+
+  // Fills block_number_for_each_element
+  for (const std::string& grid_name : grid_names) {
+    size_t end_position = grid_name.find(',', 1);
+    block_number_for_each_element.push_back(
+        static_cast<size_t>(std::stoi(grid_name.substr(2, end_position))));
+  }
+
+  auto max_block_number =
+      *std::max_element(block_number_for_each_element.begin(),
+                        block_number_for_each_element.end());
+  auto number_of_blocks = max_block_number + 1;
+  sorted_element_indices.reserve(number_of_blocks);
+
+  // Properly sizes subvectors of sorted_element_indices
+  for (size_t i = 0; i < number_of_blocks; ++i) {
+    std::vector<size_t> sizing_vector;
+    auto grid_names_in_block =
+        static_cast<size_t>(std::count(block_number_for_each_element.begin(),
+                                       block_number_for_each_element.end(), i));
+    sizing_vector.reserve(grid_names_in_block);
+    sorted_element_indices.push_back(sizing_vector);
+  }
+
+  // Organizing grid_names by block
+  for (size_t i = 0; i < block_number_for_each_element.size(); ++i) {
+    sorted_element_indices[block_number_for_each_element[i]].push_back(i);
+  }
+
+  return std::make_tuple(number_of_blocks, block_number_for_each_element,
+                         sorted_element_indices);
+}
+
+// Sort input by block
+template <typename T>
+std::vector<std::vector<T>> sort_by_block(
+    const std::vector<std::vector<size_t>>& sorted_element_indices,
+    const std::vector<T>& property_to_sort) {
+  std::vector<std::vector<T>> sorted_property;
+  sorted_property.reserve(sorted_element_indices.size());
+
+  // Properly sizes subvectors
+  for (const auto& sorted_block_index : sorted_element_indices) {
+    std::vector<T> sizing_vector;
+    sizing_vector.reserve(sorted_block_index.size());
+    for (const auto& sorted_element_index : sorted_block_index) {
+      sizing_vector.push_back(property_to_sort[sorted_element_index]);
+    }
+    sorted_property.push_back(sizing_vector);
+  }
+
+  return sorted_property;
+}
+
+// Returns properties for each block
```

Comment:
Neat, but could you explain what you mean by `properties`? :D Generally, this function is a bit more technical since the size of the connectivity is very important, so I think here expanding a bit on the algorithm and the reason for it would be useful to future users and devs :)